lib/Config/Configuration.cpp

```diff
@@ -36,47 +43,108 @@ PhasarConfig::PhasarConfig() {
+}
+
+std::unique_ptr<llvm::MemoryBuffer>
+PhasarConfig::readConfigFile(const llvm::Twine &FileName) {
+  return getOrThrow(readConfigFileOrErr(FileName));
+}
+
+std::string PhasarConfig::readConfigFileAsText(const llvm::Twine &FileName) {
+  auto Buffer = readConfigFile(FileName);
+  return Buffer->getBuffer().str();
+}
+
+llvm::ErrorOr<std::unique_ptr<llvm::MemoryBuffer>>
+PhasarConfig::readConfigFileOrErr(const llvm::Twine &FileName) {
+  if (auto LocalConfigPath = LocalConfigurationDirectory()) {
+    if (llvm::sys::fs::exists(*LocalConfigPath + FileName)) {
+      PHASAR_LOG_LEVEL(INFO,
+                       "Local config file: " << (*LocalConfigPath + FileName));
+      return readFileOrErr(*LocalConfigPath + FileName);
+    }
+  }
+  PHASAR_LOG_LEVEL(INFO, "Global config file: "
+                             << (GlobalConfigurationDirectory() + FileName));
+  return readFileOrErr(GlobalConfigurationDirectory() + FileName);
+}
+llvm::ErrorOr<std::string>
+PhasarConfig::readConfigFileAsTextOrErr(const llvm::Twine &FileName) {
+  return mapValue(readConfigFileOrErr(FileName),
+                  [](auto Buffer) { return Buffer->getBuffer().str(); });
+}
+
+std::unique_ptr<llvm::MemoryBuffer>
+PhasarConfig::readConfigFileOrNull(const llvm::Twine &FileName) {
+  return getOrEmpty(readConfigFileOrErr(FileName));
+}
+std::optional<std::string>
+PhasarConfig::readConfigFileAsTextOrNull(const llvm::Twine &FileName) {
+  if (auto Buffer = readConfigFileOrNull(FileName)) {
+    return Buffer->getBuffer().str();
+  }
+  return std::nullopt;
+}
+
 void PhasarConfig::loadGlibcSpecialFunctionNames() {
-  const std::string GLIBCFunctionListFilePath =
-      (ConfigurationDirectory() + GLIBCFunctionListFileName).str();
-
-  if (std::filesystem::exists(GLIBCFunctionListFilePath)) {
-    // Load glibc function names specified in the config file
-    std::vector<std::string> GlibcFunctions;
-    std::string Glibc = readTextFile(GLIBCFunctionListFilePath);
-    // Insert glibc function names
-    boost::split(GlibcFunctions, Glibc, boost::is_any_of("\n"),
-                 boost::token_compress_on);
-
-    SpecialFuncNames.insert(GlibcFunctions.begin(), GlibcFunctions.end());
-  } else {
+  auto Glibc = readConfigFileAsTextOrErr(GLIBCFunctionListFileName);
+  if (!Glibc) {
+    if (Glibc.getError() != std::errc::no_such_file_or_directory) {
+      PHASAR_LOG_LEVEL(WARNING, "Could not open config file '"
+                                    << GLIBCFunctionListFileName
+                                    << "': " << Glibc.getError().message());
+    }
     // Add default glibc function names
     SpecialFuncNames.insert({"_exit"});
+    return;
   }
+
+  llvm::SmallVector<llvm::StringRef, 0> GlibcFunctions;
+  llvm::SplitString(*Glibc, GlibcFunctions, "\n");
+
+  llvm::transform(GlibcFunctions,
+                  std::inserter(SpecialFuncNames, SpecialFuncNames.end()),
+                  [](llvm::StringRef Str) { return Str.trim().str(); });
```

Comment:
this is pretty similar to the parsing below. although just some lines, I think it might make sense to create a function for this.